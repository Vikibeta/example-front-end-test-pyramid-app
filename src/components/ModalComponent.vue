<template>
    <div class="modal" :class="{'is-active': displayModal}">
      <div class="modal-background"></div>
        <div class="modal-card">
          <header class="modal-card-head">
            <p class="modal-card-title">The front-end test pyramid</p>
            <button class="delete" aria-label="close" @click="toggleModal"></button>
          </header>
          <section class="modal-card-body">
            The front-end test pyramid is a guide for organizing test suites.
          </section>
          <footer class="modal-card-foot">
            <button class="button is-success" @click="toggleModal">Ok</button>
          </footer>
        </div>
  </div>
</template>

<script>
import { mapGetters, mapActions } from 'vuex'

export default {
  name: 'modal-component',
  computed: mapGetters([
    'displayModal'
  ]),
  methods: mapActions([
    'toggleModal'
  ])
}
</script>
