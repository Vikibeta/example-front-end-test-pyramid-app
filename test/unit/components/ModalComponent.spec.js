import { createLocalVue, shallow } from 'vue-test-utils'
import { createRenderer } from 'vue-server-renderer'
import Vuex from 'vuex'
import ModalComponent from '../../../src/components/ModalComponent'

describe('ModalComponent.vue', () => {
  test('has class is-active when displayModal is true', () => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    const store = new Vuex.Store({
      getters: {displayModal: () => true},
      actions: {
        toggleModal: () => {}
      }
    })
    const wrapper = shallow(ModalComponent, {
      localVue,
      store
    })
    expect(wrapper.element.classList).toContain('is-active')
  })

  test('has class is-active when displayModal is false', () => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    const store = new Vuex.Store({
      getters: {displayModal: () => false},
      actions: {
        toggleModal: () => {}
      }
    })
    const wrapper = shallow(ModalComponent, {
      localVue,
      store
    })
    expect(wrapper.element.classList).not.toContain('is-active')
  })

  test('calls toggleModal when delete button is clicked', () => {
    const actions = {
      toggleModal: jest.fn()
    }
    const localVue = createLocalVue()
    localVue.use(Vuex)
    const store = new Vuex.Store({
      actions,
      getters: {displayModal: () => {}}
    })
    const wrapper = shallow(ModalComponent, {
      localVue,
      store
    })
    wrapper.find('.delete').trigger('click')
    expect(actions.toggleModal).toHaveBeenCalled()
  })

  test('calls toggleModal when success button is clicked', () => {
    const actions = {
      toggleModal: jest.fn()
    }
    const localVue = createLocalVue()
    localVue.use(Vuex)
    const store = new Vuex.Store({
      actions,
      getters: {displayModal: () => {}}
    })
    const wrapper = shallow(ModalComponent, {
      localVue,
      store
    })
    wrapper.find('.is-success ').trigger('click')
    expect(actions.toggleModal).toHaveBeenCalled()
  })

    it('has same HTML structure', () => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    const store = new Vuex.Store({
      actions: {toggleModal: () => {}},
      getters: {displayModal: () => {}}
    })
    const renderer = createRenderer()
    const wrapper = shallow(ModalComponent, {
      localVue,
      store
    })
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
