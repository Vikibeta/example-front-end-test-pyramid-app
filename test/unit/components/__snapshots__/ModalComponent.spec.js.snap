// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalComponent.vue has same HTML structure 1`] = `
<div class="modal">
    <div class="modal-background"></div>
    <div class="modal-card">
        <header class="modal-card-head">
            <p class="modal-card-title">The front-end test pyramid</p> <button aria-label="close" class="delete"></button></header>
        <section class="modal-card-body">
            The front-end test pyramid is a guide for organizing test suites.
        </section>
        <footer class="modal-card-foot"><button class="button is-success">Ok</button></footer>
    </div>
</div>
`;
